src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -39,7 +39,17 @@ public static SocketAsyncEngine Instance
                         {
                             var engine = new SocketAsyncEngine(SocketAsyncEngineBackend.Create());
                             Task.Factory.StartNew(o => {
-                                ((SocketAsyncEngine)o)._backend.EventLoop();
+                                for (;;)
+                                {
+                                    try
+                                    {
+                                        ((SocketAsyncEngine)o)._backend.EventLoop();
```

Comment:
Consider moving the cast to be before the for loop so as to avoid needing to do it repeatedly (I realize we should only loop around again at this point if there was an exception, but still :smile:)